internal/provider/resource/integration_aws_secrets_manager.go

```diff
@@ -444,14 +458,36 @@ func (r *IntegrationAWSSecretsManagerResource) Update(ctx context.Context, req r
 		return
 	}
 
-	var planOptions AwsSecretsManagerOptions
+	if authMethod == pkg.AwsAuthMethodAccessKey {
+		updateIntegrationAuthRequest.AccessId = plan.AccessKeyID.ValueString()
+		updateIntegrationAuthRequest.AccessToken = plan.SecretAccessKey.ValueString()
+	} else if authMethod == pkg.AwsAuthMethodAssumeRole {
+		updateIntegrationAuthRequest.AWSAssumeIamRoleArn = plan.AssumeRoleArn.ValueString()
+	}
 
```

Comment:
I noticed that we already have a `RequiresReplace` modifier for these fields. Would this make any difference? Same is the case for the other integration resources 
<img width="996" alt="image" src="https://github.com/user-attachments/assets/d46982b1-3495-4154-b0bb-d5cb6313e923">

Reply:
Need to connect with @DanielHougaard over this one.